src/lib/viewers/image/ImageViewer.js

```diff
@@ -513,6 +514,15 @@ class ImageViewer extends ImageBaseViewer {
      */
     handleAnnotationControlsClick({ mode }) {
         const nextMode = this.annotationControlsFSM.transition(AnnotationInput.CLICK, mode);
+
+        if (nextMode === AnnotationMode.REGION) {
```

Comment:
We should avoid duplicating logic between files. Can this be moved to a reusable location?

Reply:
Yes I can move it to BaseViewer.js and have both children share a parent method.